Dynatrace-Environment-ServiceLevelObjective/src/handlers.ts

```diff
@@ -13,28 +26,98 @@ type SlosPayload = {
     slo: SloPayload[]
 } & PaginatedResponseType;
 
-class Resource extends AbstractDynatraceResource<ResourceModel, SloPayload, SloPayload, void, TypeConfigurationModel> {
+type EventuallyConsistentCallbackContext = {
+    serverTiming?: number
+} & RetryableCallbackContext;
+
+class Resource extends AbstractDynatraceResource<ResourceModel, AxiosResponse<SloPayload>, AxiosResponse<SloPayload>, void, TypeConfigurationModel> {
 
     private userAgent = `AWS CloudFormation (+https://aws.amazon.com/cloudformation/) CloudFormation resource ${this.typeName}/${version}`;
+    private callbackDelay = 30;
+
+    @handlerEvent(Action.Update)
+    async updateHandler(
+        session: Optional<SessionProxy>,
+        request: ResourceHandlerRequest<ResourceModel>,
+        callbackContext: EventuallyConsistentCallbackContext,
+        logger: LoggerProxy,
+        typeConfiguration: TypeConfigurationModel
+    ): Promise<ProgressEvent<ResourceModel, EventuallyConsistentCallbackContext>> {
+        let model = this.newModel(request.desiredResourceState);
+
+        if (!(await this.assertExists(model, typeConfiguration))) {
+            throw new exceptions.NotFound(this.typeName, request.logicalResourceIdentifier);
+        }
+
+        if (callbackContext.retry && callbackContext.retry > this.maxRetries) {
+            throw new exceptions.NotStabilized(`Resource failed to stabilized after ${this.maxRetries} retries`);
+        }
+
+        if (!callbackContext.serverTiming) {
+            try {
+                await this.update(model, typeConfiguration);
+                let getResponse = await this.get(model, typeConfiguration);
+                let serverTiming = this.getServerTiming(getResponse);
+
+                let progressEventBuilder = ProgressEvent.builder<ProgressEvent<ResourceModel, EventuallyConsistentCallbackContext>>()
+                    .status(OperationStatus.InProgress)
+                    .resourceModel(model)
+                    .callbackContext({
+                        serverTiming: serverTiming,
+                        retry: 1
+                    });
+                if (!serverTiming) {
+                    progressEventBuilder.callbackDelaySeconds(this.callbackDelay);
+                }
+                return progressEventBuilder.build();
+            } catch (e) {
+                logger.log(`Error ${e}`);
+                this.processRequestException(e, request);
+            }
+        }
+
+        try {
+            const getResponse = await this.get(model, typeConfiguration);
+            const getServerTiming = this.getServerTiming(getResponse);
```

Comment:
It looks like we make this same call on line 60. Could we just get the `serverTiming` variable once and re-use that?

Reply:
@joshuadeanhall This seems to not be available on all API calls. I would prefer to have this into the `AbstractDynatraceResource` but looks like we can't right now.

Reply:
Well actually, I've coded this defencively so I can add `getServerTiming` method to `AbstractDynatraceResource`. Let me do that

Reply:
@joshuadeanhall Done